### PR TITLE
bigquery: proper handle options to Job.getQueryResults

### PR DESCRIPTION
**Pull request by pongad**

### CODE REVIEW

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java [hunk 1]
```diff
@@ -272,13 +275,40 @@ public TableResult getQueryResults(QueryResultsOption... options)
     }
 
     TableId table = ((QueryJobConfiguration) getConfiguration()).getDestinationTable();
-    // TODO(pongad): merge options?
+
+    List<QueryResultsOption> waitOptions = new ArrayList<>();
+    waitOptions.add(QueryResultsOption.pageSize(0L));
+    List<TableDataListOption> listOptions = new ArrayList<>();
```
Comment (pongad) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java]: I'm somewhat concerned by this. If/When we eventually move to streaming RPC, most of these options will be useless.

Comment (tswast) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java]: That's going to be a problem across languages. We'll figure out how to transition if/when we get a streaming RPC API for this.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java [hunk 2]
```diff
@@ -272,13 +275,40 @@ public TableResult getQueryResults(QueryResultsOption... options)
...
+          listOptions.add(TableDataListOption.startIndex((Long) option.getValue()));
+          break;
+        case TIMEOUT:
+          hasWaitTime = true;
+          waitOptions.add(QueryResultsOption.maxWaitTime((Long) option.getValue()));
+          break;
+      }
+    }
+    if (!hasWaitTime) {
+      waitOptions.add(QueryResultsOption.maxWaitTime(Duration.ofMinutes(1).toMillis()));
```
Comment (tswast) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java]: I think the API default is 10 minutes. Is there a reason to explicitly set it if it's not provided?

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java (hunk 1 above)
Comment (pongad) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java]: SGTM

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java (hunk 2 above)
Comment (pongad) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java]: I did it to keep existing behavior, but you're right that it seems odd to second-guess the API. Removed.

**Pull request merged**